src/framebuffer.rs

```diff
@@ -567,7 +584,35 @@ impl Framebuffer {
         let r = color & 0xff;
         let g = (color >> 8) & 0xff;
         let b = (color >> 16) & 0xff;
-        _ = write!(dst, "\x1b[{typ}8;2;{r};{g};{b}m");
+
+        // If the terminal doesn't support truecolor, the rgb channel needs to be
+        // downsampled to display properly.
+        match self.color_mode {
+            ColorMode::TrueColor => {
+                _ = write!(dst, "\x1b[{typ}8;2;{r};{g};{b}m");
+            }
+            ColorMode::Color256 => {
+                let index: u8 = if r == g && g == b {
```

Comment:
I think the grayscale detection is a little brittle. It may be better to convert the 216+24 colors to Oklab and store them in a list here. Then we can search linearly through it and find the closest coordinate and its index.

While writing a 4D matcher I found that Rust is entirely unable to optimize `[f32; 4]` operations from MOVSS to MOVPS under `opt-level=s` which is quite disappointing to learn. I suppose I should hurry my custom stdlib collection reimplementation so we can switch back to `opt-level=3` sooner rather than later.

All that aside, I would prefer if the entire code is only compiled under `cfg(target_platform = "macos")`.